thrust/system/detail/generic/shuffle.inl

```diff
@@ -119,7 +147,7 @@ class feistel_bijection {
   uint64_t left_side_bits;
   uint64_t right_side_mask;
   uint64_t left_side_mask;
-  uint64_t key[num_rounds];
+  uint64_t key[num_rounds * 2];
```

Comment:
Where does this 2 come from?

Reply:
Just because the wyhash function I used took two keys. In my original testing I found that you needed at least 128 bits of randomness to consistently random results